src/capture.js

```diff
@@ -328,6 +328,15 @@ Capture.setElementContentFromString = function(el, htmlString) {
         // <head> content should be everything before this.
         captured.headContent = rawHTML.slice(0, match.index);
 
+        // Ensure that there are no comment tags with <head> in them,
+        // or the following setup will cause problems:
+        //
+        // <html>
+        // <head foo="bar">
+        // <!-- <head> begin -->
+        // <plaintext>
+        captured.headContent = captured.headContent.replace(new RegExp('<!--[\\s\\S]*<head>[^>]*-->'), '');
```

Comment:
I worry that this is too specific, since instances like this will not match:

`<!--<head class="blah">-->`
`<!--
blah blah blah
<head>-->`
<!--<head>-->

Playground: http://www.regexpal.com/?fam=93812

Thoughts @jansepar?